master/master.cfg

```diff
@@ -73,6 +73,35 @@ def do_step_build_spl(step):
 def do_step_build_zfs(step):
     return do_step_build(step, 'buildzfs')
 
+style_factory = util.BuildFactory()
+
+style_factory.addStep(Git(repourl=zfs_repo, workdir="build/zfs",
+    mode="full", method="clobber", codebase="zfs",
+    logEnviron=False, getDescription=True,
+    description=["cloning"], descriptionDone=["cloned"]))
+style_factory.addStep(ShellCommand(command=["./autogen.sh"],
+    env={'PATH' : bin_path},
+    workdir="build/zfs", logEnviron=False,
+    decodeRC={0 : SUCCESS, 1 : FAILURE, 2 : WARNINGS, 3 : SKIPPED },
+    description=["autogen"], descriptionDone=["autogen"]))
+style_factory.addStep(ShellCommand(command=["./configure", "--with-config=srpm"],
+    env={'PATH' : bin_path},
+    workdir="build/zfs", logEnviron=False,
+    decodeRC={0 : SUCCESS, 1 : FAILURE, 2 : WARNINGS, 3 : SKIPPED },
+    description=["configuring"], descriptionDone=["configure"]))
```

Comment:
I'd prefer to use the `bb-build-zfs.sh` script for autogen.sh and configure for consistency with the other builds.  It looks like the script should just work if you set the `CONFIG_OPTIONS="--with-config=srpm"` in the environment of the buildstep.  This will cause `make` to be run but since nothing gets compiled it'll be fast.  Plus we'll get some test coverage on the build system functionality.  You should be able to use the zfs build step from the build factory with minimal modification.